### ISSUE TITLE
Wrong path for single files

### ISSUE THREAD

**Issue by klodoma** [paul-thebaud/phpunitgen-console#2]
If I run these 2 commands, the generated files land in different folder structure. 
```
phpunitgen src\Models\Import
phpunitgen src\Models\Import\ImagesImport.php
```

See attached screenshot:
![Screenshot_115](https://user-images.githubusercontent.com/944350/98090326-2a5a2800-1e84-11eb-817d-46125f78e6b9.png)

phpunitgen.json:
```
{
  "overwriteFiles": false,
  "backupFiles": true,
  "excludedFiles": [],
  "includedFiles": [
    "\\.php$"
  ],
  "generateOnMake": true,
  "automaticGeneration": true,
  "implementations": {
    "PhpUnitGen\\Core\\Contracts\\Generators\\TestGenerator": "PhpUnitGen\\Core\\Generators\\Tests\\DelegateTestGenerator"
  },
  "baseNamespace": "App",
  "baseTestNamespace": "Tests",
  "testCase": "App\\Tests\\TestCase",
  "excludedMethods": [
    "__construct",
    "__destruct"
  ],
  "mergedPhpDoc": [
    "author",
    "copyright",
    "license",
    "version"
  ],
  "phpDoc": [],
  "options": {
    "context": "laravel"
  }
}
```

**Comment by paul-thebaud** [comment]
Hello,
Sorry for the long delay, I missed the notification.
Looking into this soon.

**Comment by paul-thebaud** [comment]
Hi!
I tried to reproduce this problem using PhpUnitGen, without success.
Could you please create a GitHub repository or Gist with a reproduction of the problem?
In advance, thanks.

**Comment by klodoma** [comment]
> Hi!
> I tried to reproduce this problem using PhpUnitGen, without success.
> Could you please create a GitHub repository or Gist with a reproduction of the problem?
> In advance, thanks.

Sure. I will create a repo.

**Comment by klodoma** [comment]
Here is the repo:

https://github.com/klodoma/phpunitgen-issue

I used these 2 commands:
```
phpunitgen src\Models\Import
phpunitgen src\Models\Import\ImportCsv.php
```
Ok, I've just tested it on a linux machine and it seems in happens only on Windows.

![Screenshot_127](https://user-images.githubusercontent.com/944350/99193890-1237b200-277c-11eb-93e2-63d83f14c0e8.png)
![Screenshot_128](https://user-images.githubusercontent.com/944350/99193891-12d04880-277c-11eb-9682-b40218e5d4fc.png)

**Comment by paul-thebaud** [comment]
Hi @klodoma
Thanks for the repository.
I'm on MacOS, that's why I can reproduce I suppose.
I'll try on Windows in the week to found where the problem is. Probably due to sources or targets resolvers.

**Comment by klodoma** [comment]
Ok, I've digged around and the problem is here:
https://github.com/paul-thebaud/phpunitgen-console/blob/6f061a57489924247a6ccde093004bb506d0db87/src/Files/TargetResolver.php#L48

Of you parse a directory you get paths with '/' where as if you specify a file, the '\' separator is used and this function fails. 
Probably it would make sense, to replace all '\' with '/' in the resolver.

**Comment by klodoma** [comment]
```
    /**
     * {@inheritdoc}
     */
    public function resolve(ClassFactory $classFactory, string $sourcePath, string $targetPath): string
    {
        if ($targetPath === 'tests' && $classFactory->getTestSubNamespace() !== '') {
            $targetPath .= '/'.str_replace('\\', '/', $classFactory->getTestSubNamespace());
        }
        $targetPath = preg_replace('/\/+/', '/', $targetPath.'/');
        $sourcePath = str_replace("\\", '/', $sourcePath);

        $compiledPath = Str::replaceFirst($this->filesystem->getRoot(), '', $sourcePath);
        $compiledPath = $targetPath.Str::afterFirst('/', $compiledPath);

        return Str::replaceLast('.php', 'Test.php', $compiledPath);
    }
```

Should I make a pull request?

**Comment by paul-thebaud** [comment]
Hi @klodoma!

Sorry again for the delay, I took a look into the issue, and your work helped me, thanks!
I'm not using your PR (but thanks for it) because the "path fix" should be done inside the source resolver and file system, so all other operations can be done on correct paths. It was already managed on file system, that's why for a directory there is no problem. The problem comes from the SourceResolver not cleaning the path for a single file.
I added documentation on Filesystem and SourceResolver methods to explain that paths should be using "/" as the dir separator.
I also added a test on the source resolver for this case, and added the unit/feature tests compatibility on Windows.

**I released the 1.3.2 version with the bug fix.**

I let you update the version on your side and test. Please close this issue if everything is OK. Feel free to comment/create an issue if there is another problem.

Have a nice day! :)

**Comment by klodoma** [comment]
Look good. Thank you!
